zipkin-local.js

```diff
@@ -0,0 +1,148 @@
+const _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
+const {
+  PLUGIN_OPTIONS,
+} = require("gatsby-plugin-newrelic-test/utils/constants");
+const getCiData = require("./utils/getCiData");
+const _zipkin = _interopRequireDefault(require("zipkin"));
+const _zipkinTransportHttp = require("zipkin-transport-newrelic");
+const _zipkinJavascriptOpentracing = _interopRequireDefault(
+  require("zipkin-javascript-opentracing")
+);
+const _nodeFetch = _interopRequireDefault(require("node-fetch"));
+const {now} = require('./utils/time');
+
+const ciAttributes = getCiData();
+
+let recorder, 
+    logger;
+/**
+ * Create and return an open-tracing compatible tracer. See
+ * https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.ts
+ */
+
+const create = () => {
+  const { staging, NR_INGEST_KEY, traces: tags = {}, SITE_NAME } = PLUGIN_OPTIONS;
+  logger = new _zipkinTransportHttp.HttpLogger({
+    endpoint: `https://${
+      staging ? `staging-` : ``
+    }trace-api.newrelic.com/trace/v1`,
+    jsonEncoder: _zipkin.jsonEncoder.JSON_V2,
+    httpInterval: 2000,
+    headers: {
+      "Api-Key": NR_INGEST_KEY,
+      "Data-Format": "zipkin",
+      "Data-Format-Version": 2,
+      options: PLUGIN_OPTIONS,
+      tags: {
+        ...tags,
+        ...ciAttributes,
+      },
+    },
+  });
+  recorder = new _zipkin.BatchRecorder({
+    logger,
+    // timeout = 60 hours, must be longer than site's build time
+    timeout: 60 * 60 * 60 * 1000000,
```

Comment:
For readability, this might be nice as a constant at the top of the file:

```js
// 60 hours, must be longer than site's build time
const TIMEOUT_LENGTH = 60 * 60 * 60 * 1000000;
```